api/internal/localizer/localizer_test.go

```diff
@@ -753,3 +753,58 @@ nameSuffix: -test
 	addFiles(t, fSysExpected, "/localized", kustAndComponents)
 	checkFSys(t, fSysExpected, fSys)
 }
+
+func TestLocalizeResources(t *testing.T) {
```

Comment:
Could we add a tests here with a remote resource file, and a remote resource directory?

Reply:
We can't add that test yet because the code that calculates the path to save a remote root, #4909, hasn't been merged yet.

Also, since these are integration tests, I planned to add them in a separate PR in `krusty`. I can add them in `krusty` in this PR once #4909 is merged?

Reply:
I thought that the `localizeFile` and `localizeDir` functions are supposed to be converting the remote URLs to local filepaths, so IMO these tests should cover that behavior. Can we get #4909 in and then add the tests here?

Reply:
I'm also OK to merge this before #4909, if we add the tests in #4909. Let me know which you prefer.

Reply:
> I thought that the `localizeFile` and `localizeDir` functions are supposed to be converting the remote URLs to local filepaths, so IMO these tests should cover that behavior. Can we get #4909 in and then add the tests here?

Sure, sg.